ModularTegustation/tegu_mobs/lc13_outskirtdwellers.dm

```diff
@@ -447,3 +447,598 @@
+	icon = 'ModularTegustation/Teguicons/32x48.dmi'
+	icon_state = "lovetown_shambler"
+	icon_living = "lovetown_shambler"
+	icon_dead = "lovetown_shambler"
+	maxHealth = 900
+	health = 900
+	move_to_delay = 6
+	ranged = TRUE
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 1.4, WHITE_DAMAGE = 0.6, BLACK_DAMAGE = 0.8, PALE_DAMAGE = 2)
+	move_resist = MOVE_FORCE_OVERPOWERING
+	mob_spawn_amount = 4 //on death explodes into 4 suicidals
+
+	var/scream_cooldown
+	var/scream_cooldown_time = 7 SECONDS
+	var/can_act = TRUE
+
+/mob/living/simple_animal/hostile/lovetown/shambler/AttackingTarget()
+	return OpenFire()
+
+/mob/living/simple_animal/hostile/lovetown/shambler/Move()
+	if(!can_act)
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/lovetown/shambler/OpenFire()
+	if(scream_cooldown <= world.time && prob(85))
+		Scream()
+	return
+
+/mob/living/simple_animal/hostile/lovetown/shambler/proc/Scream()
+	can_act = FALSE
+	scream_cooldown = world.time + scream_cooldown_time
+	playsound(get_turf(src), 'sound/creatures/lc13/lovetown/scream.ogg', 75, TRUE, 3)
+	for(var/i = 1 to 3)
+		var/obj/effect/temp_visual/decoy/D = new /obj/effect/temp_visual/decoy(get_turf(src), src)
+		animate(D, alpha = 0, transform = matrix()*1.5, time = 2)
+		SLEEP_CHECK_DEATH(3)
+	for(var/mob/living/L in view(6, src))
+		if(!faction_check_mob(L))
+			L.apply_damage(40, WHITE_DAMAGE, null, L.run_armor_check(null, WHITE_DAMAGE))
+	can_act = TRUE
+
+// Love Town Slumberer - HE threat,, quite damaging and can grab you, stuns you for some time.
+/mob/living/simple_animal/hostile/lovetown/slumberer
+	name = "love town slumberer"
+	desc = "A muscular mass of flesh that has two very long arms capable of reaching far"
+	icon = 'ModularTegustation/Teguicons/64x64.dmi'
+	icon_state = "lovetown_slumberer"
+	icon_living = "lovetown_slumberer"
+	icon_dead = "lovetown_slumberer"
+	base_pixel_x = -16
+	pixel_x = -16
+	maxHealth = 1000
+	health = 1000
+	melee_damage_lower = 30
+	melee_damage_upper = 35
+	melee_damage_type = RED_DAMAGE
+	armortype = RED_DAMAGE
+	rapid_melee = 0.5
+	attack_sound = 'sound/creatures/lc13/lovetown/slam.ogg'
+	attack_verb_continuous = "grapples"
+	attack_verb_simple = "grapple"
+	move_to_delay = 10 //Absurdly slow
+	ranged = TRUE
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 1.4, WHITE_DAMAGE = 0.6, BLACK_DAMAGE = 0.8, PALE_DAMAGE = 2)
+	move_resist = MOVE_FORCE_OVERPOWERING
+	mob_spawn_amount = 2
+
+	var/grab_ready = FALSE //are we ready to grab
+	var/countering = FALSE //are we grabbing them right now
+	var/counter_threshold = 400 //2 counters at most
+	var/damage_taken
+
+/mob/living/simple_animal/hostile/lovetown/slumberer/proc/DisableCounter()
+	if(!countering)
+		icon_state = icon_living
+		grab_ready = FALSE
+
+/mob/living/simple_animal/hostile/lovetown/slumberer/proc/CounterGrab(target)
+	countering = TRUE
+	grab_ready = FALSE
+	playsound(get_turf(src), attack_sound , 10, 3, 3)
+	var/mob/living/carbon/human/H = target
```

Comment:
If you're going to assume the target is a carbon, simply due that in the proc definition and add an istype check afterwards.
```suggestion
/mob/living/simple_animal/hostile/lovetown/slumberer/proc/CounterGrab(mob/living/carbon/human/H)
	if(!istype(H))
		return
	countering = TRUE
	grab_ready = FALSE
	playsound(get_turf(src), attack_sound , 10, 3, 3)
```

Reply:
Sorry, didnt know, changes done